ledfx/effects/random.py

```diff
@@ -0,0 +1,91 @@
+
+
+class RandomEffect(TemporalEffect):
+    """
+    Randomly activates sections
+    """
+
+    NAME = "Random"
+    CATEGORY = "Non-Reactive"
+
+    CONFIG_SCHEMA = vol.Schema(
+        {
+            vol.Optional(
+                "hit_color", description="Hit color", default="#FFFFFF"
+            ): validate_color,
+            vol.Optional(
+                "hit_duration",
+                description="Hit duration",
+                default=0.1,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.1, max=5.0)),
+            vol.Optional(
+                "hit_probability_per_sec",
+                description="Probability of hit per second",
+                default=0.1,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0.01, max=1.0)),
+            vol.Optional(
+                "hit_relative_size",
+                description="Hit size relative to LED strip",
+                default=10,
+            ): vol.All(vol.Coerce(int), vol.Range(min=1, max=100)),
+        }
+    )
+
+    def __init__(self, ledfx, config):
+        super().__init__(ledfx, config)
+        self.last_time = timeit.default_timer()
+
+    def config_updated(self, config):
+        self.hit_color = np.array(
+            parse_color(self._config["hit_color"]), dtype=float
+        )
+        self.hit_relative_size = self._config["hit_relative_size"]
+        self.hit_duration = self._config["hit_duration"]
+        self.speed = self._config["speed"]
+        self.probability_per_sec = (
+            self.__balance_hit_probability_based_on_speed()
+        )
+
+    def on_activate(self, pixel_count):
+        self.last_time = timeit.default_timer()
+
+    def effect_loop(self):
+        hit_absolute_size = int(
+            self.pixel_count * self.hit_relative_size / 100
+        )
+
+        # handle time variant
+        now = timeit.default_timer()
+        time_passed = now - self.last_time
+
+        hit_is_still_active = time_passed < self.hit_duration
+        if not hit_is_still_active:
+            # Create a base frame of all-off (black) pixels
+            self.pixels = np.zeros((self.pixel_count, 3), dtype=np.float64)
+
+            is_hit = np.random.random() < self.probability_per_sec
+            if is_hit:
+                # assign pixel hit at random position
+                random_pos = random.randrange(
+                    self.pixel_count - hit_absolute_size + 1
+                )
+                # frame slice based of random_pos will be hit
+                self.pixels[random_pos : random_pos + hit_absolute_size] = (
+                    np.tile(self.hit_color, (hit_absolute_size, 1))
+                )
+                self.last_time = timeit.default_timer()
+
+    def __balance_hit_probability_based_on_speed(self) -> float:
+        runs_per_sec = self.speed * 10
+        # this is the probability per effect run
+        return 1 - (1 - self._config["hit_probability_per_sec"]) ** (
```

Comment:
What is the difference between speed and hit probability per sec, shouldn't this just be one param?

Reply:
I find it to be rather unintuitive to use the speed param to model the hit probability. Of course it's possible though. Do you have some adaption in mind how to map the speed param to a hit probability?

Reply:
I think this is just a case of semantics. If both sliders are only used for this one thing, then it only needs a single slide with a good name.

Speed just looks like a multiplier against probability. So end user shouldn't have two sliders to care about.

I need to look at this from a PC again, I might be missing something from phone